### DISCUSSION TITLE
Streaming Joint Positions to a KUKA robot in Grasshopper

### DISCUSSION THREAD

**Discussion by Jishnu8** [visose/Robots#212] [Q&A] [answered]
Currently, I have managed to load a Kuka robot (AA - KR30) using the Robots plugin and moved it to a particular fixed joint position (or multiple joint positions) in the simulation using Grasshopper. 
My goal however is to continuously stream joint positions (ideally from a C++ file) to a Kuka robot (AA - KR30) which is loaded in grasshopper using the Robots plugin. Similarly, I would also like to send the current joint positions of the Kuka robot in the simulation (in Rhino) to this same C++ file. The idea I had to accomplish the following task is to create a UPD client/server in C++ which could somehow continuously update the joint positions the Kuka robot has to move to in the Grasshopper simulation and also get the current joint positions. But I have no clue how to get started with this. 
Could anyone help me with this or suggest a way to accomplish my goal?

**Comment by visose** [comment]
Hi @Jishnu8, could you explain your objective from a high level? It looks like you are describing the attempted solution but would be good to understand the context, see: https://xyproblem.info/

I'm not sure I understand "streaming joint positions from a C++ file". Usually a C++ code file would contain text in the form of programming code that has to be compiled. Not sure if you want a text file format used to transfer data, such as json.

**Comment by Jishnu8** [comment]
Hi @visose, really sorry. I should have given a high-level overview of the problem first. 
I am currently developing a simple simulator for a Kuka robot in Grasshopper. The goal is to be able to continuously send joint positions which are being constantly updated from a source to Grasshopper and make the robot move to these joint positions in the simulation. Conversely, I would also like to send the current joint positions of the robot in Grasshopper to this source.
Currently, this "source" is a C++ codebase in which the target joint positions are calculated in some way (not important). So I would like to create a C++ script by which I can send these joint positions to Grasshopper continuously and ensure that the robot tries to move to these joint positions.
Now coming to the idea I had to achieve this goal. To send joint positions from a C++ file to Grasshopper, the idea I had was to create a Upd server script in C++ that would continuously send these joint positions. However, I am not sure how to go about creating a Udp client within Grasshopper itself which would continuously receive the message the server is sending.
Is the objective clearer now? 
Thank you so much for your time again.

**Comment by visose** [comment, marked as answer]
I think you said the same thing 😆. It would be interesting to know what the purpose of this C++ program is, why you need to visualize them in Grasshopper if you are creating these robot poses in something unrelated to Rhino.

In any case, the solution you are proposing regarding feeding data into Grasshopper from an outside source seems unrelated to the Robots plugin. You will have more luck finding for info about this if you search on the Grasshopper forums, without including the robots aspect.
